File: frontend/cli/src/app/designaciones/designacion.ts
import { Cargo } from '../cargos/cargo';
import { Persona } from '../personas/persona';

export interface Designacion {
    id: number;
    situacionRevista: string;
    fechaInicio: string; 
    fechaFin?: string | null;
    cargo: Cargo;
    persona: Persona;
  }

File: backend/src/main/java/unpsjb/labprog/backend/model/Designacion.java
package unpsjb.labprog.backend.model;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Designacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(length = 45)
    private String situacionRevista;

    private LocalDate fechaInicio;
    private LocalDate fechaFin;

    @ManyToOne
    private Cargo cargo;

    @ManyToOne
    private Persona persona;
}
